lib/src/script.dart

```diff
@@ -381,9 +382,9 @@ class _ScriptSpan extends SourceSpanMixin implements FileSpan {
   SourceSpan union(SourceSpan other) {
     if (other is! _ScriptSpan) return super.union(other);
 
-    var span = expand(other);
-    var beginSpan = span._startPosition == _startPosition ? this : other;
-    var endSpan = span._endPosition == _endPosition ? this : other;
+    _ScriptSpan span = expand(other);
+    _ScriptSpan beginSpan = span._startPosition == _startPosition ? this : other;
+    _ScriptSpan endSpan = span._endPosition == _endPosition ? this : other;
```

Comment:
You should be able to get rid of most of these by making the `is!` check above into an `is` check that encompasses all of these.